### ISSUE TITLE
Installed VSCode plugin but still not able to catch deno syntax

### ISSUE THREAD

**Issue by zemse** [denoland/vscode_deno#48]
Following the tutorial in the docs:
```
const res = await fetch("https://jsonplaceholder.typicode.com/todos/1");
const body = new Uint8Array(await res.arrayBuffer());
await Deno.stdout.write(body);
```
I still get underline on the await:
```
'await' expressions are only allowed at the top level of a file when that file is a module, but this file has no imports or exports. Consider adding an empty 'export {}' to make this file a module.ts(1375)
```
And underline on the Deno
```
Cannot find name 'Deno'.ts(2304)
```
while if I run the file with deno, it obv runs. But vscode underlines are very irritating as well as it is not giving any type suggestions for `Deno` object.

I installed the plugin Deno `1.32.0` by justjavac. I tried doing [command]+[p] and `> Enable Deno`, which says deno is already enabled.

I tried to find any similar issue but seems it's not there while my issue is very basic regarding installation. I went through Readme but not able to solve the problem. Can you please help me what more I need to do to get it working?

**Comment by zemse** [comment]
Hey, I just uninstalled justjavac's `1.32.0` and installed axetroy's `3.7.0` and it's working wonderful as expected. What's the matter with justjavac version? It's got like 16k downloads while 6k on axetroy's. How come so many people don't get this error?

**Comment by justjavac** [comment]
@zemse Can you provide a tsserver.log file, I will fix this bug

**Comment by zemse** [comment]
Hi, here is the file. [tsserver.log](https://github.com/denoland/vscode_deno/files/4667771/tsserver.log).

**Comment by zemse** [comment]
I just called `> Deno: Enable Deno` on your plugin, and it created a `.vscode` dir containing 
```
{
  "deno.enable": true
}
```
I did try this yesterday, enable and disable a number of times while `.vscode` dir wasn't being created.

But now it is being created, making the plugin's work. Looks like there isn't any issue from the plugin side and maybe this mac just went silly and needed a restart :( My bad, pardon for the trouble. Closing the issue.

**Comment by zemse** [comment]
@justjavac I've run into same problem again, I have `"deno.enable": true,` in the settings.json file but still it appears like if the plugin is not active. I tried disabling Deno using `> Deno: Disable Deno` and re-enabling it again. Also observed that enabling and disabling flips `deno.enable` boolean value too. Can you please look into my [tsserver.log](https://github.com/denoland/vscode_deno/files/4669676/tsserver.log) file?

**Comment by justjavac** [comment]
```
Info 30   [23:42:6.187] Loading global plugin typescript-deno-plugin
Info 31   [23:42:6.187] Enabling plugin typescript-deno-plugin from candidate paths: /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features,/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js/../../..
Info 32   [23:42:6.187] Loading typescript-deno-plugin from /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features (resolved to /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/node_modules)
Info 33   [23:42:6.198] Loading typescript-deno-plugin from /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js/../../.. (resolved to /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/node_modules)
Info 34   [23:42:6.203] Failed to load module 'typescript-deno-plugin' from /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/node_modules: Error: Could not resolve JS module 'typescript-deno-plugin' starting at '/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/node_modules'. Looked in: /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/node_modules/typescript-deno-plugin/package.json, /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/node_modules/typescript-deno-plugin.js, /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/node_modules/typescript-deno-plugin.jsx, /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/node_modules/typescript-deno-plugin/index.js, /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/node_modules/typescript-deno-plugin/index.jsx, /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript-deno-plugin/package.json, /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript-deno-plugin.js, /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript-deno-plugin.jsx, /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript-deno-plugin/index.js, /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript-deno-plugin/index.jsx, /Applications/Visual Studio Code.app/Contents/Resources/app/node_modules/typescript-deno-plugin/package.json, /Applications/Visual Studio Code.app/Contents/Resources/app/node_modules/typescript-deno-plugin.js, /Applications/Visual Studio Code.app/Contents/Resources/app/node_modules/typescript-deno-plugin.jsx, /Applications/Visual Studio Code.app/Contents/Resources/app/node_modules/typescript-deno-plugin/index.js, /Applications/Visual Studio Code.app/Contents/Resources/app/node_modules/typescript-deno-plugin/index.jsx, /Applications/Visual Studio Code.app/Contents/Resources/node_modules/typescript-deno-plugin/package.json, /Applications/Visual Studio Code.app/Contents/Resources/node_modules/typescript-deno-plugin.js, /Applications/Visual Studio Code.app/Contents/Resources/node_modules/typescript-deno-plugin.jsx, /Applications/Visual Studio Code.app/Contents/Resources/node_modules/typescript-deno-plugin/index.js, /Applications/Visual Studio Code.app/Contents/Resources/node_modules/typescript-deno-plugin/index.jsx, /Applications/Visual Studio Code.app/Contents/node_modules/typescript-deno-plugin/package.json, /Applications/Visual Studio Code.app/Contents/node_modules/typescript-deno-plugin.js, /Applications/Visual Studio Code.app/Contents/node_modules/typescript-deno-plugin.jsx, /Applications/Visual Studio Code.app/Contents/node_modules/typescript-deno-plugin/index.js, /Applications/Visual Studio Code.app/Contents/node_modules/typescript-deno-plugin/index.jsx, /Applications/Visual Studio Code.app/node_modules/typescript-deno-plugin/package.json, /Applications/Visual Studio Code.app/node_modules/typescript-deno-plugin.js, /Applications/Visual Studio Code.app/node_modules/typescript-deno-plugin.jsx, /Applications/Visual Studio Code.app/node_modules/typescript-deno-plugin/index.js, /Applications/Visual Studio Code.app/node_modules/typescript-deno-plugin/index.jsx, /Applications/node_modules/typescript-deno-plugin/package.json, /Applications/node_modules/typescript-deno-plugin.js, /Applications/node_modules/typescript-deno-plugin.jsx, /Applications/node_modules/typescript-deno-plugin/index.js, /Applications/node_modules/typescript-deno-plugin/index.jsx, /node_modules/typescript-deno-plugin/package.json, /node_modules/typescript-deno-plugin.js, /node_modules/typescript-deno-plugin.jsx, /node_modules/typescript-deno-plugin/index.js, /node_modules/typescript-deno-plugin/index.jsx
    at Object.resolveJSModule (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:29284:19)
    at Object.sys.require (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:147642:45)
    at Function.Project.resolveModule (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:139196:35)
    at /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:140205:36
    at Object.firstDefined (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:197:26)
```

This extension will first look for typescript-deno-plugin in typescript-language-features, then look for it in justjavac.vscode-deno.

Judging from the log, it didn't look in the justjavac.vscode-deno installation directory, and reported an error directly.

When you open the ts file, did you see the Deno extension information in the status bar?

![](https://user-images.githubusercontent.com/359395/82780581-3c25b100-9e8a-11ea-9d24-b5c839dff251.png)

**Comment by justjavac** [comment]
If it is a problem with the plugin installation, you can press `Cmd + Shift + P`, enter "openexlog", you will see the log file `exthost.log`

**Comment by zemse** [comment]
I do see Deno 1.0.2. While I was getting this too. Once again I checked that `deno.enable` is true in `.vscode/settings.json`. 
This is the file [exthost.log](https://github.com/denoland/vscode_deno/files/4676107/exthost.log).

![Screenshot 2020-05-25 13 56 09](https://user-images.githubusercontent.com/22412996/82794363-fec42200-9e8f-11ea-8684-f1818616fc63.png)

Few months ago in a different project, I did some setting in vscode to pick typescript version in `./node_modules/typescript` in the vscode because a bug was fixed in latest typescript version and I wanted it instead of default typescript of vscode. Is it possible that this might be causing the problem? However, when I open a deno project, then I get a notification that vscode is using default typescript because it couldn't find the node_modules one.

**Comment by teroyks** [comment]
> And underline on the Deno
> 
> ```
> Cannot find name 'Deno'.ts(2304)
> ```

I have this same issue in VS Code. I can see 'Deno 1.0.2' on the status bar, and other Deno syntax (such as import, await) seems to work (VS Code doesn't report problems on them, only on the Deno name so far).

**Comment by iMarv** [comment]
Getting the same issue with the "official" `denoland.Deno`, but got it working with the deprecated `jusjavac.Deno`.

The `denoland.Deno` extension seems to be completely outdated compared to the `justjavac.Deno`/
Do they share the same version numbers or is `denoland.Deno@1.24.0` === `justjavac.Deno@1.34.0`?

**Comment by justjavac** [comment]
@iMarv denoland.vscode-deno@1.24.0 **!==** justjavac.vscode-deno@1.34.0

justjavac.vscode-deno@1.34.0 is the master branch of denoland/vscode_deno

**Comment by ry** [comment]
Closed by https://github.com/denoland/vscode_deno/pull/77

**Comment by petekp** [comment]
I'm setting up Deno for the first time in a new project and the VSCode extension host log shows this error:

```
[2020-08-30 14:35:30.746] [exthost] [error] Activating extension denoland.vscode-deno failed due to an error:
[2020-08-30 14:35:30.753] [exthost] [error] Error: Cannot get typescript APIs. try restart Visual Studio Code.
    at getTypescriptAPI (/Users/petepetrash/.vscode/extensions/denoland.vscode-deno-2.0.16/client/out/client/src/extension.js:31:17)
    at Extension.activate (/Users/petepetrash/.vscode/extensions/denoland.vscode-deno-2.0.16/client/out/client/src/extension.js:268:28)
    at Function._callActivateOptional (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:877:509)
    at Function._callActivate (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:877:160)
    at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:875:492
    at processTicksAndRejections (internal/process/task_queues.js:85:5)
    at async Promise.all (index 2)
```

Restarting VSCode and re-installing the extension did not solve the issue. Using VSCode 1.48.2 and tsc@4.0.2.

**Comment by TranXuanHoang** [comment]
I got the same phenomenal in which Deno code suggestion and autocompletion did not work after installing `denoland`'s [vscode_deno](https://github.com/denoland/vscode_deno/). After trying around I found the following steps help VS Code understands Deno - though [this isn't a configuration recommended by the plugin's official documentation](https://github.com/denoland/vscode_deno#configuration)

- Create a directory named `.vscode` and then create a `settings.json` file inside that directory. Put the following JSON config in that file

    `your-project-workspace/.vscode/settings.json`
    ```json
    {
      "deno.enable": true,
    }
    ```

- Open VS Code Settings `File > Preferences > Settings` (or `Ctr + ,` on Windows / `Cmd + ,` on macOS). Search `deno`, then select `Extensions > Deno` fron the search result filter.
![image](https://user-images.githubusercontent.com/4038986/93851360-ea335180-fcea-11ea-84ef-b84c310facc8.png)

- Check the `Controls whether Deno is enabled or not` to enable Deno like the above image capture.
- Restart VS Code, and hopefully Deno suggestion and autocompletion will work
![image](https://user-images.githubusercontent.com/4038986/93851606-662d9980-fceb-11ea-84ea-bc091931010a.png)

**Comment by merima98** [comment]
Thank you TranXuanHoang! It was so helpful to me.

**Comment by kmschott** [comment]
Perhaps `init deno` will solve the problem: In `vscode` "show all commands" via CMD/CTRL + SHIFT + P and search for `init deno`, run it. Helped me to solve all problems with shown errors in vscode with deno.

**Comment by Timmmm** [comment]
Hmm I still have this issue, and I have enabled the Deno extension for this workspace and restarted VSCode.

![image](https://user-images.githubusercontent.com/376842/118359078-afcc3c80-b579-11eb-9565-6507f497d702.png)

With Deno 1.10.1.

**Comment by Timmmm** [comment]
Ah it seems like you have to do `deno run <script.ts>` in order for Deno to download the imports. That kind of sucks; other language extensions offer to do that for you (e.g. Dart).

Also the `ts` error is still there so it seems like the Deno extension isn't disabling Typescript support as it says it should here:

![image](https://user-images.githubusercontent.com/376842/118359152-118ca680-b57a-11eb-8beb-dd36fc22a251.png)

**Comment by Timmmm** [comment]
Ah you can manually set this in your workspace settings:

```
"typescript.validate.enable": false,
"javascript.validate.enable": false,
```

Still it would be nice if the Deno extension did that for you as it claims to.

**Comment by filipe-freire** [comment]
> Perhaps `init deno` will solve the problem: In `vscode` "show all commands" via CMD/CTRL + SHIFT + P and search for `init deno`, run it. Helped me to solve all problems with shown errors in vscode with deno.

Just tried the solution mentioned here, and it worked 💯. Thanks for sharing!

**Comment by Stokestack** [comment]
I seemed to have the Deno extension working in VS Code, but console.log() was sending nothing to the console. Checking online, I created a launch.json file that included "outputCapture": "std".

After that, the Run/Debug pane on the left changed its appearance. It no longer has the two big buttons at the top, but rather a "Launch program" drop-down at the top and some sections for variables, watch, call stack. Upon pressing "Launch program," I get

`error: Module not found "file:///Users/me/data/Mustache_test/main.ts".`

But nothing calls for main.ts, which I've never had in this directory. I assume that this means the Deno integration isn't working (I'm brand-new to Deno and JS/TS development). I re-ran the "configure this workspace for Deno" option from the command palette, which claimed success. No change.
